Fixing and udpating the GDPR procedure (#1627)
codenvy/che-docs: modules/administration-guide/partials/proc_removing-user-data.adoc

> 

Reviewer: ```suggestion
```

> 

Reviewer: ```suggestion
```

> :parent-context-of-removing-user-data: {context}

Reviewer: ```suggestion
```

> 

Reviewer: ```suggestion
```

[id="removing-user-data_{context}"]
= Removing user data

> 

Reviewer: ```suggestion
```

> :context: removing-user-data

Reviewer: ```suggestion
```

== Removing user data according to GDPR

> The General Data Protection Regulation (GDPR) law enforces the right for individuals to have personal data erased.

Reviewer: Add a link to some source that explains what GDPR is.

Author: Done. Updated with the main GDPR Wiki link.

The following procedure describes how to remove a user’s data from a cluster and the {identity-provider} database.

[NOTE]
====

> * Following commands use the default namespace, `_{prod-namespace}_`, as a user's namespace example after the `-n` option.

Reviewer: ```suggestion
Following commands use the default namespace, `{prod-namespace}`, as a user's namespace example after the `-n` option.
```

====

.Prerequisites

> * A user or an administrator authorization token. If a user wants to delete any other data except the data bound to their user account, the `admin` privileges are required.

Reviewer: ```suggestion
* A user or an administrator authorization token. To delete any other data except the data bound to a user account, `admin` privileges are required. The `admin` is a special {prod-short} administrator account pre-created and enabled using the `pass:[CHE_SYSTEM_ADMIN__NAME]` and `pass:[CHE_SYSTEM_SUPER__PRIVILEGED__MODE = true]` Custom Resource definitions.
```

> +

Reviewer: ```suggestion
```

> The `admin` is a  special {prod-short} administrator account pre-created and enabled using the:

Reviewer: ```suggestion
```

> +

Reviewer: ```suggestion
```

> [subs="+quotes,macros"]

Reviewer: ```suggestion
```

> * `pass:[CHE_SYSTEM_ADMIN__NAME]`

Reviewer: ```suggestion
```

> [subs="+quotes,macros"]

Reviewer: ```suggestion
```

> * `pass:[CHE_SYSTEM_SUPER__PRIVILEGED__MODE = true]` 

Reviewer: ```suggestion
```

> +

Reviewer: ```suggestion
```

> Custom Resource definitions.

Reviewer: ```suggestion
```

+
[subs="+quotes,macros"]
----
spec:
 server:
   customCheProperties:
     pass:[CHE_SYSTEM_SUPER__PRIVILEGED__MODE]: 'true'
     pass:[CHE_SYSTEM_ADMIN__NAME]: '__<admin-name>__'

>      

Reviewer: ```suggestion
```

----
+

> If needed, use commands below for creating an `admin` user:

Reviewer: ```suggestion
If needed, use commands below for creating the `admin` user:
```

+
[subs="+quotes,macros,attributes"]
----
$ oc patch checluster {prod-checluster} \
  --type merge \

>   -p '{ "spec": { "server": {"customCheProperties": pass:[{"CHE_SYSTEM_SUPER__PRIVILEGED__MODE": "true"}] } }}' \

Reviewer: ```suggestion
  -p '{ "spec": { "server": {"customCheProperties": {"pass:[CHE_SYSTEM_SUPER__PRIVILEGED__MODE]": "true"} } }}' \
```

  -n __{prod-namespace}__
----
+
[subs="+quotes,macros,attributes"]
----
$ oc patch checluster {prod-checluster} \
  --type merge \

>   pass:[-p '{ "spec": { "server": {"customCheProperties": {"CHE_SYSTEM_ADMIN__NAME": "__<admin-name>__"}] } }}' \

Reviewer: The pass-through macro must be used only the things you don't want interpreted. Your code results in this:
![image](https://user-images.githubusercontent.com/6623103/95356394-e694f280-08c6-11eb-8e75-ec30c14cb887.png)
(I.e. review locally before pushing :)
```suggestion
  -p '{ "spec": { "server": {"customCheProperties": {"pass:[CHE_SYSTEM_ADMIN__NAME]": "__<admin-name>__"} } }}' \
```

.Authorization token privileges:

* `admin` - Can delete all personal data of all users
* `user` - Can delete only the data related to the user
====

* A user or an administrator is logged in the OpenShift cluster with deployed {prod-short}.

* A user ID is obtained. Get the user ID using the commands below:

** For the current user:
+
[subs="+quotes,macros,attributes"]
----
$ curl -X GET \
  --header 'Authorization: Bearer __<user-token>__' \
  'pass:c,m,a,q[+https:+//__<{prod-host}>__]/api/user'
----
// {prod-url-secure}/api/user - vs - http(s)://{prod-host}/api/user
 
** To find a user by name: 
+
[subs="+quotes,macros,attributes"]
----
$ curl -X GET \
  --header 'Authorization: Bearer __<user-token>__' \
  'pass:c,m,a,q[+https:+//__<{prod-host}>__]/api/user/find?name=__<username>__'
----
 
** To find a user by email: 
+
[subs="+quotes,macros,attributes"]
----
$ curl -X GET \
  --header 'Authorization: Bearer __<user-token>__' \
  'pass:c,m,a,q[+https:+//__<{prod-host}>__]/api/user/find?email=__<email>__'
----

> 

Reviewer: ```suggestion
+
```

.Example of obtaining a user ID

> 

Reviewer: ```suggestion
+
```

This example uses `vparfono` as a local user name.

> 

Reviewer: ```suggestion
+
```

====
[subs="+quotes,macros,attributes"]
----
$ curl -X GET \
  --header 'Authorization: Bearer __<user-token>__' \
  'https://che-vp-che.apps.che-dev.x6e0.p1.openshiftapps.com/api/user/find?name=vparfono'
----

The user ID is at the bottom of the curl command output.

----
{
 "name": "vparfono",
 "links": [
   {
.
.
.
   }
 ],

>  "email": "vitaly.parfonov@gmail.com",

Reviewer 2: ```suggestion
 "email": "vparfono@redhat.com",
```

Author: will edit

 "id": "921b6f33-2657-407e-93a6-fb14cf2329ce"
}
----
====

.Procedure

> . Update the `customresourceCR` file to permit the removal of a user’s data from the {identity-provider} database:

Reviewer: Is the file really called `customresourceCR`?

Author: Updated to:

. Update the `{prod-checluster}` `CheCluster Custom` Resource (CR) definition to permit the removal of a user’s data from the {identity-provider} database:

+
[subs="+quotes,macros,attributes"]
----
$ oc patch checluster/{prod-checluster} \

>   --patch "{\"spec\":{\"server\":{\"customCheProperties\": pass:[{\"CHE_KEYCLOAK_CASCADE__USER__REMOVAL__ENABLED\"]: \"true\"}}}}" \

Reviewer: ```suggestion
  --patch "{\"spec\":{\"server\":{\"customCheProperties\": {\"pass:[CHE_KEYCLOAK_CASCADE__USER__REMOVAL__ENABLED]\": \"true\"}}}}" \
```

>   --type=merge -n __{prod-namespace}__

Reviewer: ```suggestion
  --type=merge -n _{prod-namespace}_
```

----

. Remove the data using the API:
+
[subs="+quotes,macros,attributes"]
----
$ curl -i -X DELETE \
  --header 'Authorization: Bearer __<user-token>__' \
  pass:c,m,a,q[+https:+//__<{prod-host}>__]/api/user/__<user-id>__
----

.Verification

> The execution of the `curl -i -X DELETE --header 'Authorization: Bearer __<user-token>__' pass:c,m,a,q[+https:+//__<{prod-host}>__]/api/user/__<user-id>__` command, returns the code 204 as the API response.

Reviewer: ```suggestion
Running the following command returns code `204` as the API response:
+
 [subs="+quotes,macros,attributes"]
 ====
$ curl -i -X DELETE \
  --header 'Authorization: Bearer __<user-token>__' \
  pass:c,m,a,q[+https:+//__<{prod-host}>__]/api/user/__<user-id>__
====
```

.Additional resources

> To remove the data of all the users, follow the instructions for xref:installation-guide:uninstalling-che.adoc[].

Reviewer: ```suggestion
+
To remove the data of all users, follow the instructions for xref:installation-guide:uninstalling-che.adoc[].
```

> 

Reviewer: ```suggestion
```

> :context: {parent-context-of-removing-user-data}

Reviewer: ```suggestion
```
